cmd/snap/main_test.go

```diff
@@ -41,11 +44,59 @@ import (
 // Hook up check.v1 into the "go test" runner
 func Test(t *testing.T) { TestingT(t) }
 
+func openPty() (pty, tty *os.File, err error) {
+	p, err := os.OpenFile("/dev/ptmx", os.O_RDWR, 0)
```

Comment:
I was planning to see if this code could be trimmed down. It's just wholesale pasted from elsewhere to see if the concept works at the moment. I'm *hoping* it can be made more minimal.